src/ServiceControl.Monitoring/QueueLength/QueueLengthReportHandler.cs

```diff
@@ -1,22 +1,27 @@
 ﻿namespace ServiceControl.Monitoring.QueueLength
 {
     using System.Threading.Tasks;
+    using Infrastructure;
     using NServiceBus;
     using NServiceBus.Metrics;
 
     class QueueLengthReportHandler : IHandleMessages<MetricReport>
     {
+        EndpointRegistry endpointRegistry;
         QueueLengthDataStore queueLengthDataStore;
 
-        public QueueLengthReportHandler(QueueLengthDataStore queueLengthDataStore)
+        public QueueLengthReportHandler(EndpointRegistry endpointRegistry, QueueLengthDataStore queueLengthDataStore)
         {
+            this.endpointRegistry = endpointRegistry;
             this.queueLengthDataStore = queueLengthDataStore;
         }
 
         public Task Handle(MetricReport message, IMessageHandlerContext context)
         {
             var endpointInstanceId = EndpointInstanceId.From(context.MessageHeaders);
 
+            endpointRegistry.Record(endpointInstanceId);
```

Comment:
I've added dedicated handler with explicit declaration of messages. I think it will make it less fragile if for some reason we start getting messages which are not metric reports.